primehub/deployments.py

```diff
@@ -42,16 +42,28 @@ def invalid_config(message: str):
     raise PrimeHubException(explain)
 
 
+def auto_gen_id(name: str):
+    import re
+    import random
+    normalized_name = re.sub(r'[\W_]', '-', name).lower()
+    random_string = str(random.random())[2:7]
+    return f'{normalized_name}-{random_string}'
+
+
 def verify_requires(config):
     # verify required fields in the config
     if not config:
         invalid_config('Deployment definition is required.')
+    if 'name' not in config:
+        invalid_config('name is required')
     if 'id' not in config:
-        invalid_config('id is required')
+        config['id'] = auto_gen_id(config['name'])
```

Comment:
Please don't mix the validation and auto-filling feature 

The auto-filling feature must be done before `validation`, so we could validate the inputs even someone users or our auto-filling feature put wrong values.